x-pack/plugins/security_solution/public/management/cypress/e2e/endpoint/response_console.cy.ts

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Agent } from '@kbn/fleet-plugin/common';
+import {
+  inputConsoleCommand,
+  openResponseConsoleFromEndpointList,
+  performCommandInputChecks,
+  submitCommand,
+  waitForCommandToBeExecuted,
+  waitForEndpointListPageToBeLoaded,
+} from '../../tasks/response_console';
+import type { IndexedFleetEndpointPolicyResponse } from '../../../../../common/endpoint/data_loaders/index_fleet_endpoint_policy';
+import {
+  getAgentByHostName,
+  getEndpointIntegrationVersion,
+  reassignAgentPolicy,
+} from '../../tasks/fleet';
+import { checkEndpointListForIsolatedHosts, createAgentPolicyTask } from '../../tasks/isolate';
+import { login } from '../../tasks/login';
+import { ENDPOINT_VM_NAME } from '../../tasks/common';
+
+describe('Response console', () => {
+  const endpointHostname = Cypress.env(ENDPOINT_VM_NAME);
+
+  beforeEach(() => {
+    login();
+  });
+
+  describe('Isolate command', () => {
+    let response: IndexedFleetEndpointPolicyResponse;
+    let initialAgentData: Agent;
+
+    before(() => {
+      getAgentByHostName(endpointHostname).then((agentData) => {
+        initialAgentData = agentData;
+      });
+
+      getEndpointIntegrationVersion().then((version) => {
+        createAgentPolicyTask(version, (data) => {
+          response = data;
+        });
+      });
+    });
+
+    after(() => {
+      if (initialAgentData?.policy_id) {
+        reassignAgentPolicy(initialAgentData.id, initialAgentData.policy_id);
+      }
+      if (response) {
+        cy.task('deleteIndexedFleetEndpointPolicies', response);
+      }
+    });
+
+    it('should isolate host from response console', () => {
+      waitForEndpointListPageToBeLoaded(endpointHostname);
+      checkEndpointListForIsolatedHosts(false);
+      openResponseConsoleFromEndpointList();
+      performCommandInputChecks('isolate');
+      submitCommand();
+      waitForCommandToBeExecuted();
+    });
+
+    it('should release host from response console', () => {
+      waitForEndpointListPageToBeLoaded(endpointHostname);
+      checkEndpointListForIsolatedHosts(true);
+      openResponseConsoleFromEndpointList();
+      performCommandInputChecks('release');
+      submitCommand();
+      waitForCommandToBeExecuted();
+      waitForEndpointListPageToBeLoaded(endpointHostname);
+      checkEndpointListForIsolatedHosts(false);
+    });
```

Comment:
I do agree, that we should aim to create independent test cases, however, in this particular case `describe` are independent and tests are not :). Reason for that is sub sequential tests expect state that is set in previous tests (to unisolate a host we have to isolate it first and so on) and I see two approaches:

1. We do everything in one test case (get processes, save one pid, kill the pid, get processes, check if pid changed, suspend pid)
2. We repeat all these actions in each test case. 

What I did, is I went with `1.` but divided the checks for readability. Im open to suggestions and ideas though :)

Reply:
I see your point, and also just realised that this is real e2e test and not a mocked one (like the one I linked above), so it would be much-much slower to re-create the starting states for every test case, so I agree with you! 👍

One suggestion, to make this decision more visible for our future selves: maybe we could name the `describe` blocks as user journeys., e.g. `User Journey: Isolate command` or `User Journey for Isolate command: isolating and releasing an endpoint` or so. This would indicate that these are not really independent test cases but rather a user journey split up to smaller tasks. Let me know your thoughts.

Reply:
Sounds good!